### PR TITLE
new Kubeapps Dashboard

### PR DESCRIPTION
**Pull request by prydonius**
- [x] adds apprepository-controller
- [x] also updates kube.libsonnet to latest version for CRD support
- [x] add k8s API proxy to ingress so the dashboard can manage apprepos and other objects
- [x] add chartsvc
- [x] updates dashboard to use chartsvc (https://github.com/kubeapps/dashboard/pull/14)
- [x] kubernetes 1.8 becomes default in GKE

### CODE REVIEW

File: apprepository.jsonnet [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  },
+
+  _apprepo(name, url):: kube._Object("kubeapps.com/v1alpha1", "AppRepository", name) + $.namespace {
+    spec: {
+      url: url,
+      type: "helm"
+    },
+  },
+
+  apprepos: {
```
Comment (prydonius) [inline apprepository.jsonnet:80]: this adds the default stable and incubator repos, I have a later plan to not do this and instead have a setup wizard in the dashboard UI that recommends chart repos to add

**Comment by prydonius** [comment]
@anguslees any ideas why the tests are failing for RBAC? It's not clear to me, and I haven't seen any issues locally.

**Comment by anguslees** [comment]
> @anguslees any ideas why the tests are failing for RBAC?

I don't see anything RBAC related in the following, so it's possible I'm looking at the wrong log:
https://travis-ci.org/kubeapps/manifest/builds/326789615?utm_source=github_status&utm_medium=notification

What I see in that log is the custom resources failing for two different reasons: 1: the version of kubecfg in use isn't reliably sorting the CRD before the custom resource (fixed by using a newer version), and 2: we're trying to use the CR before the CRD controller has had time to actually create the resource kind.

The latter is a bit tricky, since it requires kubecfg (or something) to treat CRD resources specially and wait until they have done their thing before proceeding with other resources.

**Comment by prydonius** [comment]
> I don't see anything RBAC related in the following

Yeah this was what threw me off, as it appeared the non-RBAC test in the matrix was passing but the RBAC ones were failing.

So it sounds like this is the same issue that https://github.com/kubeapps/kubeapps/pull/112 resolved in Kubeapps. We could also update the CI to build kubeapps with the updates manifests.

**Pull request merged**